platform/android/java_godot_lib_jni.cpp

```diff
@@ -237,6 +244,7 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_step(JNIEnv *env, jcl
 		// Since Godot is initialized on the UI thread, _main_thread_id was set to that thread's id,
 		// but for Godot purposes, the main thread is the one running the game loop
 		Main::setup2(Thread::get_caller_id());
+		input_handler = new AndroidInputHandler();
```

Comment:
Makes sense. However, I first tried to construct the input handler at `Java_org_godotengine_godot_GodotLib_initialize()`. However, it seems that's too soon for `InputDefault` to be created and is causing a flood of errors and the game is unable to start. Regarding the fact it's created on the render thread and used from the UI one, shouldn't be a problem, since the `step` variable issues memory barriers on increment and comparison. Not that I like that approach much, though, so if you can suggest a good place to move the construction of the input handler to, I'd listen to you.

Reply:
Good point. Looking at the previous logic, `InputDefault` was initialized for `OS_Android` in `OS_Android::initialize(...)` which itself is called in `Main::setup2(...)`, so where you placed the init logic is actually correct.

The only alternative I can think of is to not store a reference to `InputDefault` in the class and always just use `InputDefault::get_singleton()` instead, but that seems overly bothersome..

I'll leave the decision to you, the current location is fine by me.

Reply:
Alright then. Let's leave it this way. We can always improve it later; this is likely to be revisited.